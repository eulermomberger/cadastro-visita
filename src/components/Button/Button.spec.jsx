import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react';

import { Button } from '.';

describe('Button component', () => {
  it('Should appear with the correct background color', () => {
    const testRenderer = renderer.create(
      <Button
        title='Bot達o'
        backgroundColor='#064482'
        onClick={() => {}}
      />
    );

    try {
      expect(testRenderer.toJSON()).toMatchSnapshot();
    } finally {
      testRenderer.unmount();
    }
  });

  it('Should appear with the default background color', () => {
    const testRenderer = renderer.create(
      <Button
        title='Bot達o'
        onClick={() => {}}
      />
    );

    try {
      expect(testRenderer.toJSON()).toMatchSnapshot();
    } finally {
      testRenderer.unmount();
    }
  });

  it('Should call the function prop when clicked', () => {
    const mockFunc = jest.fn();

    const { getByText } = render(
      <Button
        title='Bot達o'
        backgroundColor='#064482'
        onClick={mockFunc}
      />
    );

    fireEvent.click(getByText(/Bot達o/i));

    expect(mockFunc).toHaveBeenCalled();
  });
});
