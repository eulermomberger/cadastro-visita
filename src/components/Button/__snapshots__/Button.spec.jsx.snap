// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Should appear with the correct background color 1`] = `
<button
  className="button-container"
  onClick={[Function]}
  style={
    {
      "backgroundColor": "#064482",
    }
  }
>
  Botão
</button>
`;

exports[`Button component Should appear with the default background color 1`] = `
<button
  className="button-container"
  onClick={[Function]}
  style={
    {
      "backgroundColor": "#6933FF",
    }
  }
>
  Botão
</button>
`;
